src/lib/operators/jit_operator/operators/jit_read_tuples.cpp

```diff
@@ -1,12 +1,71 @@
 #include "jit_read_tuples.hpp"
 
 #include "../jit_types.hpp"
+#include "all_type_variant.hpp"
 #include "expression/evaluation/expression_evaluator.hpp"
+#include "jit_expression.hpp"
 #include "resolve_type.hpp"
+#include "storage/segment_iterables/create_iterable_from_attribute_vector.hpp"
 #include "storage/segment_iterate.hpp"
 
 namespace opossum {
 
+namespace {
+
+/*
+ * AttributeIterableData holds all required data to create an iterable from an attribute vector of a dictionary
```

Comment:
Isn't this missing the information that we want a *filtered* iterable? Also, please add why we can't create the iterable right away. Finally, please add that nullptr has a semantic beyond "not set".